source/common/upstream/outlier_detection_impl.cc

```diff
@@ -280,13 +280,14 @@ DetectorImpl::create(const Cluster& cluster,
                      const envoy::config::cluster::v3::OutlierDetection& config,
                      Event::Dispatcher& dispatcher, Runtime::Loader& runtime,
                      TimeSource& time_source, EventLoggerSharedPtr event_logger) {
-  if (config.max_ejection_time() < config.base_ejection_time()) {
+  std::shared_ptr<DetectorImpl> detector(
+      new DetectorImpl(cluster, config, dispatcher, runtime, time_source, event_logger));
+
+  if (detector->config().maxEjectionTimeMs() < detector->config().baseEjectionTimeMs()) {
```

Comment:
Would it also be possible to increase the default value of max ejection time? Currently there isn't a way to override the value using the v2 API.

Reply:
The main reason behind `max_ejection_time` was to limit how much this value may grow. It determines how quickly the node comes back to the pool.
V2 is fatal in 1.17 by default.
If you insist on using V2, you can overwrite max_ejection_time using runtime values: https://www.envoyproxy.io/docs/envoy/latest/configuration/upstream/cluster_manager/cluster_runtime.html

Reply:
@hennna I'm surprised you want to increase the default. I think it's already pretty generous? In any case I agree that this is probably OK given that there is a runtime override for the field.